### PR TITLE
[syn/spi_host/usbdev] Synthesis related updates (buffer hookup points)

### PR DESCRIPTION
**Pull request by msfschaffner**
This is in preparation for introducing the CDC flow in #9031.
In particular, it removes an old and unused SDC file for earlgrey (not at the chiplevel), and it adds hand instantiated hookup points for certain signals inside `spi_host` and `usbdev` so that we can reference them properly in our SDC file (right now we rely on automatically generated synopsys DC names for these).

### CODE REVIEW

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc [hunk 1]
```diff
@@ -52,11 +52,30 @@ set USB_TCK_PERIOD [expr $USB_TCK_TARGET_PERIOD*$CLK_PERIOD_FACTOR]
...
+    set USB_PIN    gen_flops[1]*.u_size_only_reg/D
+    set USB_D_PIN  u_usb_d_flop/gen_*u_impl*/gen_flops[0]*.u_size_only_reg/Q
+    set USB_OE_PIN u_oe_flop_flop/gen_*u_impl*/gen_flops[0]*.u_size_only_reg/Q
+} else {
+    set USB_N_PIN  d_i[2]
+    set USB_P_PIN  d_i[3]
+    set USB_PIN    d_i[1]
+    set USB_D_PIN  u_usb_d_flop/gen_*u_impl*/q_o[0]
+    set USB_OE_PIN u_oe_flop_flop/gen_*u_impl*/q_o[0]
+}
```
Comment (msfschaffner) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:71]: This is in preparation for CDC.
I.e., these paths will be different depending on whether this is built with closed source LIBs or with the open-source views only.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc [hunk 2]
```diff
@@ -215,8 +234,7 @@ set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
 # During pre-layout, the SPI_HOST_CLK source latencies are estimanted to account for
 # pad and logic latencies.  After CTS, source latency must be removed as all clocks are propagated
 
-# TODO, this flop should be hand instantiated
-set CLK_DIV_PIN top_earlgrey/u_spi_host0/u_spi_core/u_fsm/sck_q_reg/Q
+set CLK_DIV_PIN top_earlgrey/u_spi_host0/u_spi_core/u_fsm/u_sck_flop/gen_*u_impl*/gen_flops[0]*.u_size_only_reg/Q
```
Comment (martin-lueker) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:237]: I see.  This looks reasonable to me.

File: hw/ip/spi_host/rtl/spi_host_fsm.sv [hunk 3]
```diff
@@ -467,21 +467,27 @@ module spi_host_fsm
...
 
-  assign sck_d = cpol ? (state_d != InternalClkHigh) :
-                        (state_d == InternalClkHigh);
+  assign sck_d = stall ? sck_q                        :
+                 cpol  ? (state_d != InternalClkHigh) :
+                         (state_d == InternalClkHigh);
 
   assign sck_o = sck_q;
 
+  prim_flop u_sck_flop (
```
Comment (martin-lueker) [inline hw/ip/spi_host/rtl/spi_host_fsm.sv:476]: Just a comment: It is interesting that our prim_flops do not have an enable signal.   If this whole block were to be implemented by hand, `stall` would be an enable signal not a combinational logic input.

This is not just about saving one gate per flop.  There are some tricky details in the implementation of the stall signal, in that it both depends on the next FSM state, but is also responsible for gating the FSM state, so there have been lots of rearrangements in this block to eliminate irreducible logic cycles.   The cycle goes like: 'A: We are ready to go to the next state. B: But there is no data, let's stall.  A: Okay, then we don't go to the next state.   B: Great, then we are not stalled."   So the stall signal becomes a ring oscillator. This gets caught by CI pretty easily though because it brings sims to a grinding halt, and Verilator catches it immediately.

Bottom Line: There are sometimes practical reasons to use the enable pin instead of a combinational data input.

**Review by martin-lueker — APPROVED** [review]

File: hw/ip/spi_host/rtl/spi_host_fsm.sv (hunk 3 above)
Comment (msfschaffner) [reply hw/ip/spi_host/rtl/spi_host_fsm.sv:476]: Thanks for pointing this out, @martin-lueker.
We do actually have a `prim_flop_en`.
I can use that one for this specific register to avoid the combo loop you have described.
Let me update the PR accordingly.

Comment (msfschaffner) [reply hw/ip/spi_host/rtl/spi_host_fsm.sv]: Updated, can you take another look @martin-lueker?

File: hw/ip/spi_host/rtl/spi_host_fsm.sv [hunk 4]
```diff
@@ -36,7 +36,7 @@ module spi_host_fsm
   input                              sw_rst_i
 );
 
-  logic             isIdle;
+  logic             is_idle;
```
Comment (martin-lueker) [inline hw/ip/spi_host/rtl/spi_host_fsm.sv:39]: Ah.  Thanks.. my bad on the naming convention here.

File: hw/ip/spi_host/rtl/spi_host_fsm.sv [hunk 5]
```diff
@@ -472,16 +472,22 @@ module spi_host_fsm
 
   assign sck_o = sck_q;
 
+  prim_flop_en u_sck_flop (
+    .clk_i,
+    .rst_ni,
+    .en_i(~stall),
```
Comment (martin-lueker) [inline hw/ip/spi_host/rtl/spi_host_fsm.sv:478]: Looks good, thanks!
